src/Component.js

```diff
@@ -46,6 +46,9 @@ export const DebounceInput = React.createClass({
 
 
   componentWillReceiveProps({value, debounceTimeout}) {
+    if (this.isDebouncing) {
```

Comment:
I will need to check if behaviour is not changed critically with this change. There are several conrollable inputs around (react-filter-input and some in projects). That might take some time.